### ISSUE TITLE
Oban.Telemetry docs don't reflect reality

### ISSUE THREAD

**Issue by eaguilera23** [sorentwo/oban#916]
### Precheck

- This might be a duplicate from: https://github.com/sorentwo/oban/issues/850

### Environment

- Oban Version: `2.15.2`
- PostgreSQL Version: `14.8`
- Elixir & Erlang/OTP Versions (`elixir --version`): `1.12.3 - OTP 23`

### Current Behavior

When a telemetry event when a job starts, the following values are passed as metadata:

```
:args
:attempt
:conf
:id
:job
:max_attempts
:prefix
:queue
:tags
:worker
```

### Expected Behavior

Based on the documentation, this is the expected metadata from a telemetry event when a job is started:

```
:conf, :job, :state
```

**Comment by sorentwo** [comment]
This is intentional.

The undocumented event fields are legacy, from before the entire job struct was passed to the event. Using those fields is deprecated, but they can't be removed without a 3.0 version bump because it's a breaking change.

**Comment by eaguilera23** [comment]
But is it intentional that the `state` key is not present at the top level?

**Comment by sorentwo** [comment]
It's intentional that `[:oban, :job, :start]` lacks a `state` key because there isn't an execution state yet. But nicely spotted that shouldn't be listed in the telemetry documentation! Fixed with d91073b
